### PR TITLE
Select's redesign

### PR DESCRIPTION
**Pull request by mvaivre**
Reduce distance to travel with cursor. Automatically detects if Popup is overflowing the window, move it accordingly.

### CODE REVIEW

**Comment by mvaivre** [comment]
> I noticed that I cannot open the popup anymore using the keyboard.

While fixing this, I've discovered that in the current version, it's impossible to select an option using the keyboard only. I'll fix it.

**Comment by mvaivre** [comment]
> I noticed that I cannot open the popup anymore using the keyboard.
While fixing this, I was struggling with the autofocus not working (ie. first option not selected on popup opening). I just realised that it's already broken on the `v2.0` branch. May be related to a layout change, I'll come back to it later.

**Comment by mvaivre** [comment]
Update: in order to fully fix accessibility and keyboard navigation, I had to stop using native `select` and `option` elements which are not really made for this kind of use.

You can now navigate through options with the arrow keys or tab key. Autofocus works as expected.

**Comment by nop33** [comment]
Since the token branch is still very WIP, maybe it'd make sense to base this branch on v2.0 instead.

I'll keep working on the tokens branch probably later tomorrow or next week and I'll address the tests.

File: src/components/Popup.tsx [hunk 1]
```diff
@@ -15,52 +15,116 @@ GNU Lesser General Public License for more details.
...
-    <Content onClick={(e) => e.stopPropagation()} role="dialog" {...fadeInOutBottomFast}>
+const minMarginToEdge = 25
+const headerHeight = 50
+
+const Popup = ({ children, onBackgroundClick, title, hookCoordinates }: PopupProps) => {
+  const { height: windowHeight, width: windowWidth } = useWindowSize() // Recompute position on window resize
+
+  const contentRef = useRef<HTMLDivElement>(null)
+
+  const [hookOffset, setHookOffset] = useState<Coordinates | undefined>(undefined)
```
Comment (nop33) [inline src/components/Popup.tsx:43]: nit: I have the feeling that this is the same as:

```tsx
const [hookOffset, setHookOffset] = useState<Coordinates>()
```

**Review by nop33 — APPROVED** [review]
Nice stuff man! My functional tests pass ✅ 

One nit thing, I notice that the toggle section (ex: passphrase) now opens a bit "too fast" imo 😅

**Comment by mvaivre** [comment]
> One nit thing, I notice that the toggle section (ex: passphrase) now opens a bit "too fast" imo 😅

Yes I agree :) We'll refine those cosmetic details (more anim variations etc.) at the very end!

**Pull request merged**